### PR TITLE
[WIP] Release 2.0.0

### PR DESCRIPTION
**Pull request by mpenkov**
@piskvorky Please have a look at this branch and let me know if you're happy with it to be released as 2.0.

@gojomo I decided against skipping to 3.0, because other projects (e.g. Django) chose this approach (1.x for Python 2.7, 2.x for Python 3+).

### CODE REVIEW

**Comment by piskvorky** [comment]
2.0 looks alright but much less urgent than the [broken Gensim](https://github.com/RaRe-Technologies/gensim/issues/2788). Let's focus on that one first, as planned.

**Comment by mpenkov** [comment]
@piskvorky Part of the problem with gensim is the log statement, and smart_open 2.0 includes the fix for that.

**Comment by piskvorky** [comment]
You're right of course.

File: appveyor.yml [hunk 1]
```diff
@@ -1,31 +0,0 @@
-environment:
```
Comment (piskvorky) [inline appveyor.yml:1]: Did we give up on automated Windows testing (AppVeyor) altogether? Or is this temporary?

Comment (mpenkov) [reply appveyor.yml]: This was a mistake, I have since reverted.

**Pull request merged**